gulpfile.js

```diff
@@ -86,26 +88,44 @@ function watch(done) {
     paths.scripts.modules,
     gulp.series(scripts.bundle, browserSync.reload)
   );
-  gulp.watch(paths.styles.src, gulp.series(styles, browserSync.reloadCSS));
+  gulp.watch(
+    paths.styles.src,
+    gulp.series(styles, criticalCss, browserSync.reloadCSS)
+  );
   gulp.watch(paths.images.src, gulp.series(imageOptim, browserSync.reload));
   gulp.watch(paths.assets.src, gulp.series(assets, browserSync.reload));
   gulp.watch(
     [paths.pages.src, paths.templates.src],
-    gulp.series(metalsmith.build, browserSync.reload)
+    gulp.series(metalsmith.build, criticalCss, browserSync.reload)
   );
   done();
 }
 
+function criticalCss() {
+  return gulp
+    .src("dist/**/*.html")
+    .pipe(
+      critical({
+        base: "dist/",
+        inline: true,
+        css: ["dist/styles/style.css"],
+        ignore: ["@font-face", /url\(/]
```

Comment:
Should we also add `extract: true,` ?

Reply:
I want to run some tests on the live website. It might be a good idea with http2, but on http1 extract is going to generate an extra request per page, and according to lighthouse that has a overall negative impact of 110ms.